server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -371,47 +372,60 @@ protected void doClose() {
     @Override
     public void executeConsistentStateUpdate(Function<RepositoryData, ClusterStateUpdateTask> createUpdateTask, String source,
                                              Consumer<Exception> onFailure) {
+        executeConsistentStateUpdate((latestRepositoryData, latestRepositoryMetadata, taskListener) -> {
+            try {
+                taskListener.onResponse(createUpdateTask.apply(latestRepositoryData));
```

Comment:
In this case, the generator for the `ClusterStateUpdateTask` requires the repository metadata, the same repository metadata that must be maintained unchanged until the update task that's based on it is executed.  

I would propose that we change the signature of the `executeConsistentStateUpdate` in `Repository`, but here I only create a proxy in `BlobStoreRepository`.